pos_longpolling/static/src/js/pos_longpolling.js

```diff
@@ -238,12 +241,20 @@ odoo.define('pos_longpolling', function(require){
     Status_Widget.include({
         set_poll_status: function(selector, current_bus) {
             var element = self.$(selector);
-            if (current_bus.longpolling_connection.status) {
-                element.removeClass('oe_red');
-                element.addClass('oe_green');
+            if (current_bus.channels.length) {
+                if (current_bus.longpolling_connection.status) {
+                    element.removeClass('oe_red');
+                    element.removeClass('oe_gray');
+                    element.addClass('oe_green');
+                } else {
+                    element.removeClass('oe_green');
+                    element.removeClass('oe_gray');
+                    element.addClass('oe_red');
+                }
             } else {
+                element.removeClass('oe_red');
                 element.removeClass('oe_green');
-                element.addClass('oe_red');
+                element.addClass('oe_gray');
```

Comment:
It must be gray when polling doesn't work (longpolling requests are not sent). It's not the same as empty channels list

Reply:
You can simplify code:
```

//compute new_class
...
// set class
element.removeClass('oe_red oe_green oe_gray').addClass(new_class)
```